skbeam/core/roi.py

```diff
@@ -423,15 +423,33 @@ def circular_average(image, calibrated_center, threshold=0, nx=100,
         Left edge of first bin defaults to minimum value of x
     max_x : float, optional number of pixels
         Right edge of last bin defaults to maximum value of x
+    mask : mask for 2D data. Assumes 1 is non masked and 0 masked.
+        None defaults to no mask.
+    
     Returns
     -------
     bin_centers : array
         The center of each bin in R. shape is (nx, )
     ring_averages : array
         Radial average of the image. shape is (nx, ).
+
+    See Also
+    --------
+    bad_to_nan_gen : Create a mask with np.nan entries
+
+    Notes
+    -----
     """
     radial_val = utils.radial_grid(calibrated_center, image.shape, pixel_size)
 
+    if mask is None:
```

Comment:
I do not think you need this as you only slice into the data if the mask is non-trivial.